core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2429,4 +2429,18 @@ package object config {
       .version("3.4.0")
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("5s")
+
+  private[spark] val CUSTOM_SERVICE_PORT_ORIGIN =
```

Comment:
I suggest defining a port range, such as (min, max], and randomly choosing to try bind within this range.

Reply:
In some environments, the upper limit of available ports may not be 65535, for example, it may be limited to (40000, 50000]

Reply:
Thanks for your reviews, this option was considered and the current design focuses on maximising the reuse of the original logic and minimising the increase in configuration.

Reply:
If need limited to (40000, 50000], the range can be controlled by the conf of spark.port.maxRetries.

Reply:
> If need limited to (40000, 50000], the range can be controlled by the conf of spark.port.maxRetries.

I'm not sure whether using `spark.port.maxRetries` to control the port range is a good idea, which looks more like a trick